core/thread.go

```diff
@@ -608,6 +628,95 @@ func (t *Thread) pushUpdate(index BlockInfo) {
 	})
 }
 
+// readable returns whether or not this thread is readable from the
+// perspective of the given address
+func (t *Thread) readable(addr string) bool {
+	if addr == t.initiator {
+		return true
+	}
+	switch t.ttype {
+	case repo.PrivateThread:
+		return false // should not happen
+	case repo.ReadOnlyThread:
```

Comment:
nit: could use `fallthrough` here?

Reply:
I thought that as well, but turns out you can't because what would you expect `t.ttype` to be when you arrive in another case......

Reply:
ah, ok